src/bsoncxx/private/error_category.cpp

```diff
@@ -120,7 +122,7 @@ class libbson_unknown_error final : public std::error_category {
         return "libbson unknown error";
     }
 
-    std::string message(int code) const noexcept override {
```

Comment:
Yes, with -Wextra. It is pretty pedantic, but it is fair that you declared a name for a parameter but then didn't use it.